### DISCUSSION TITLE
Freeze my network

### DISCUSSION THREAD

**Discussion by Limingxing00** [open-mmlab/mmdetection#9707] [Q&A]
Hi,

Thank you for your contributions to the community!

I tried to freeze some of the parameters of the head myself.

```python 

@RUNNERS.register_module()
class IterBasedRunner(BaseRunner):
    """Iteration-based Runner.

    This runner train models iteration by iteration.
    """
    def set_frozen_all(self):
        for name, module in self.model.named_modules():
            # freeze bn
            module.eval()
            # freeze weight and bais of conv/fc/...
            for param in module.parameters():
                param.requires_grad = False

    def set_not_frozen(self, nfreeze_modules):
        for nfrozen_key in nfreeze_modules:
            for name, module in self.model.named_modules():
                # do not freeze
                if nfrozen_key in name:
                    module.train()
                    for param in module.parameters():
                        param.requires_grad = True
                        # self.logger.info("{} is not frozen".format(name))

    def train(self, data_loader, **kwargs):
        self.model.train()
        
        # I add these lines
        self.set_frozen_all()
        self.set_not_frozen(["bbox_head"])

```

- I started training on a pretrained model with a learning rate 10 times smaller.
- My loss function keeps going up, I checked other components like backbone, I'm sure the weight doesn't change, only the weight of bbox_head changes.
- I also tried different learning rates, and the result is that the loss function increases.
- I used ```find_unused_parameters = True``` in the config file

![image](https://user-images.githubusercontent.com/23522700/215973018-8257d209-4942-42c1-bafe-a811ea25d464.png)

I would like to know, is there any problem with my operation like this?
